### PR TITLE
Add shelley endpoints

### PR DESCRIPTION
**Pull request by xdzurman**

### CODE REVIEW

File: src/db-api.js [hunk 1]
```diff
@@ -3,6 +3,69 @@
...
 import type { DbApi } from 'icarus-backend'; // eslint-disable-line
 
+/**
+ * Checks if addresses array contains group address. According to Emurgo,
+ * frontend should not explicitly query group addresses, it can lead to hard
+ * to detect bugs.
+ * @param {Db Object} db
+ * @param {Array<Address>} addresses
+ */
+const hasGroupAddress = (db: Pool) => async (
```
Comment (ppershing) [inline src/db-api.js:13]: we don't want to check it this way. Instead, we should really check the addresses. I believe we can use more-or-less something like
```
isGroupAddress(addr) => isHex(addr) && (addr.startswith('04') || addr.startswith('84'))
```

File: src/db-api.js [hunk 2]
```diff
@@ -3,6 +3,69 @@
...
+  account: string,
+): Promise<ResultSet> =>
+  db.query({
+    text: delegationHistoryQuery(limit),
+    values: [account],
+  })
+
+/**
+* Gets all pools and their details. Currently the tables are joined on pool owner,
+* it bloats the left join a little bit since theres a few pools which even have the same ticker.
```
Comment (ppershing) [inline src/db-api.js:47]: huh? really? pools with same ticker? what is the ticker then good for? :-(

File: src/routes.js [hunk 3]
```diff
@@ -53,6 +57,31 @@ function validateSignedTransactionReq({ signedTx } = {}) {
...
+    throw new Error('Account is empty.')
+  }
+}
+
+/**
+ * Checks for existence of group addresses which should not be explicitly requested
+ * @param {*} db Database
+ * @param {*} Server Server Config Object
+ */
+const checkForGroupAddress = (dbApi: DbApi, { logger }: ServerConfig) => async (
```
Comment (ppershing) [inline src/routes.js:91]: technically we don't need this, frontend should ensure it is sending correct requests :-)

File: src/legacy-routes.js [hunk 4]
```diff
@@ -33,10 +34,11 @@ const combinedBalance = (transactions, addresses) => {
 const txToAddressInfo = (row) => ({
   ctbId: row.hash,
   ctbTimeIssued: moment(row.time).unix(),
-  ctbInputs: row.inputs_address.map(
-    (addr, i) => [addr, { getCoin: row.inputs_amount[i] }]),
-  ctbOutputs: row.outputs_address.map(
-    (addr, i) => [addr, { getCoin: row.outputs_amount[i] }]),
+  // return null if empty to distinguish delegations
```
Comment (ppershing) [inline src/legacy-routes.js]: Don't we have certificate info as part of the table?

File: src/db-api.js (hunk 2 above)
Comment (xdzurman) [reply src/db-api.js:47]: It is supposed to be unique according to documentation, although they somehow accepted several pull request with the same ticker

File: src/legacy-routes.js (hunk 4 above)
Comment (xdzurman) [reply src/legacy-routes.js]: I guess "distinguish" was a wrong word, I meant 'allow' since they are null and map fails.

**Pull request closed without merging**